apps/framework-cli/src/framework/typescript/parser.rs

```diff
@@ -92,17 +92,37 @@ fn extract_data_models_from_ast(ast: Module) -> Result<FileObjects, TypescriptPa
     Ok(FileObjects::new(parsed_models, enums))
 }
 
-fn enum_to_data_enum(enum_decl: &TsEnumDecl) -> DataEnum {
+fn enum_to_data_enum(enum_decl: TsEnumDecl) -> Result<DataEnum, TypescriptParsingError> {
     let name = enum_decl.id.sym.to_string();
-    let values = enum_decl
-        .members
-        .iter()
-        .map(|member| match &member.id {
+    let mut values = Vec::new();
+
+    for member in enum_decl.members {
+        let name = match &member.id {
             TsEnumMemberId::Ident(ident) => ident.sym.to_string(),
             TsEnumMemberId::Str(str) => str.value.to_string(),
-        })
-        .collect();
-    DataEnum { name, values }
+        };
+
+        let value = match member.init {
+            Some(init) => match *init {
+                Expr::Lit(Lit::Str(str)) => EnumValue::String(str.value.to_string()),
+                Expr::Lit(Lit::Num(num)) => EnumValue::Int(num.value as u8),
+                _ => {
+                    return Err(TypescriptParsingError::UnsupportedDataTypeError {
+                        type_name: format!("{:?}", init),
+                    })
+                }
+            },
+            None => {
+                return Err(TypescriptParsingError::UnsupportedDataTypeError {
+                    type_name: format!("Non mapped enums"),
+                })
+            }
```

Comment:
Do we want to handle auto-increment. The following definition has values 2,3,4,5

```
enum Direction {
  Up = 2,
  Down,
  Left,
  Right,
}
```

Without `= 2`, it'll have values 0,1,2,3

Reply:
I was struggling to have something reliable that works - but with some late changes might be easier.

Reply:
the thing that's annoying is that a user could also do

```ts
enum Direction {
  Up,
  Down,
  Left = 10,
  Right,
}
```

Do you know what will be `Right`? 11 or 2?

Reply:
Cool - I will add that